usaspending_api/spending/v2/filters/spending_filter.py

```diff
@@ -0,0 +1,331 @@
+import logging
+import urllib
+from urllib.error import HTTPError
+
+from usaspending_api.awards.models import Award
+from usaspending_api.common.exceptions import InvalidParameterException
+
+logger = logging.getLogger(__name__)
+
+
+def spending_filter(alt_set, queryset, filters, _type):
+
+    for key, value in filters.items():
+        # check for valid key
+        if value is None:
+            raise InvalidParameterException('Invalid filter: ' + key + ' has null as its value.')
+
+        key_list = ['budget_function', 'budget_subfunction', 'federal_account', 'program_activity', 'object_class',
+                    'recipient', 'award', 'award_category', 'agency', 'agency_type', 'agency_sub', 'fy']
+
+        if key not in key_list:
+            raise InvalidParameterException(key + ' filter does not exist. '
+                                                  'Valid Filters: budget_function, budget_subfunction, federal_account,'
+                                                  'program_activity, object_class, recipient, award, award_category,'
+                                                  'agency, agency_type, agency_sub fy.')
+
+        # Check _type to filter on correct set (alt_set or queryset)
+        if _type == 'recipient' or _type == 'award' or _type == 'award_category' \
```

Comment:
To save code, you can declare this list near `key_list` and check to see if `_type in` that list